File: mmupnp/src/test/java/net/mm2d/upnp/empty/EmptyServiceTest.java
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.empty;

import net.mm2d.upnp.Service;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import static org.hamcrest.Matchers.*;
import static org.junit.Assert.*;

@SuppressWarnings("NonAsciiCharacters")
@RunWith(JUnit4.class)
public class EmptyServiceTest {
    @Test
    public void getDevice() {
        final Service service = new EmptyService();
        assertThat(service.getDevice(), is(not(nullValue())));
    }

    @Test
    public void getServiceType() {
        final Service service = new EmptyService();
        assertThat(service.getServiceType(), is(not(nullValue())));
    }

    @Test
    public void getServiceId() {
        final Service service = new EmptyService();
        assertThat(service.getServiceId(), is(not(nullValue())));
    }

    @Test
    public void getScpdUrl() {
        final Service service = new EmptyService();
        assertThat(service.getScpdUrl(), is(not(nullValue())));
    }

    @Test
    public void getControlUrl() {
        final Service service = new EmptyService();
        assertThat(service.getControlUrl(), is(not(nullValue())));
    }

    @Test
    public void getEventSubUrl() {
        final Service service = new EmptyService();
        assertThat(service.getEventSubUrl(), is(not(nullValue())));
    }

    @Test
    public void getDescription() {
        final Service service = new EmptyService();
        assertThat(service.getDescription(), is(not(nullValue())));
    }

    @Test
    public void getActionList() {
        final Service service = new EmptyService();
        assertThat(service.getActionList(), is(not(nullValue())));
    }

    @Test
    public void findAction() {
        final Service service = new EmptyService();
        assertThat(service.findAction(""), is(nullValue()));
    }

    @Test
    public void getStateVariableList() {
        final Service service = new EmptyService();
        assertThat(service.getStateVariableList(), is(not(nullValue())));
    }

    @Test
    public void findStateVariable() {
        final Service service = new EmptyService();
        assertThat(service.findStateVariable(""), is(nullValue()));
    }

    @Test
    public void subscribe() {
        final Service service = new EmptyService();
        assertThat(service.subscribeSync(), is(false));
    }

    @Test
    public void subscribe1() {
        final Service service = new EmptyService();
        assertThat(service.subscribeSync(true), is(false));
    }

    @Test
    public void renewSubscribe() {
        final Service service = new EmptyService();
        assertThat(service.renewSubscribeSync(), is(false));
    }

    @Test
    public void unsubscribe() {
        final Service service = new EmptyService();
        assertThat(service.unsubscribeSync(), is(false));
    }

    @Test
    public void getSubscriptionId() {
        final Service service = new EmptyService();
        assertThat(service.getSubscriptionId(), is(nullValue()));
    }
}

File: mmupnp/src/test/java/net/mm2d/upnp/empty/EmptyServiceTest.kt
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.empty

import com.google.common.truth.Truth.assertThat
import kotlinx.coroutines.runBlocking
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

@RunWith(JUnit4::class)
class EmptyServiceTest {
    @Test
    fun getDevice() {
        val service = EmptyService
        assertThat(service.device).isNotNull()
    }

    @Test
    fun getServiceType() {
        val service = EmptyService
        assertThat(service.serviceType).isNotNull()
    }

    @Test
    fun getServiceId() {
        val service = EmptyService
        assertThat(service.serviceId).isNotNull()
    }

    @Test
    fun getScpdUrl() {
        val service = EmptyService
        assertThat(service.scpdUrl).isNotNull()
    }

    @Test
    fun getControlUrl() {
        val service = EmptyService
        assertThat(service.controlUrl).isNotNull()
    }

    @Test
    fun getEventSubUrl() {
        val service = EmptyService
        assertThat(service.eventSubUrl).isNotNull()
    }

    @Test
    fun getDescription() {
        val service = EmptyService
        assertThat(service.description).isNotNull()
    }

    @Test
    fun getActionList() {
        val service = EmptyService
        assertThat(service.actionList).isNotNull()
    }

    @Test
    fun findAction() {
        val service = EmptyService
        assertThat(service.findAction("")).isNull()
    }

    @Test
    fun getStateVariableList() {
        val service = EmptyService
        assertThat(service.stateVariableList).isNotNull()
    }

    @Test
    fun findStateVariable() {
        val service = EmptyService
        assertThat(service.findStateVariable("")).isNull()
    }

    @Test
    fun getSubscriptionId() {
        val service = EmptyService
        assertThat(service.subscriptionId).isNull()
    }

    @Test
    fun subscribeAsync() {
        val service = EmptyService
        runBlocking {
            assertThat(service.subscribe()).isFalse()
        }
    }

    @Test
    fun subscribeAsync1() {
        val service = EmptyService
        runBlocking {
            assertThat(service.subscribe(true)).isFalse()
        }
    }

    @Test
    fun renewSubscribeAsync() {
        val service = EmptyService
        runBlocking {
            assertThat(service.renewSubscribe()).isFalse()
        }
    }

    @Test
    fun unsubscribeAsync() {
        val service = EmptyService
        runBlocking {
            assertThat(service.unsubscribe()).isFalse()
        }
    }
}
